src/spreadinterp.cpp

```diff
@@ -187,20 +187,21 @@ FINUFFT_NEVER_INLINE
 void print_subgrid_info(int ndims, BIGINT offset1, BIGINT offset2, BIGINT offset3,
                         UBIGINT padded_size1, UBIGINT size1, UBIGINT size2, UBIGINT size3,
                         UBIGINT M0) {
-  printf("size1 %ld, padded_size1 %ld\n", size1, padded_size1);
+  printf("size1 %" PRIu64 ", padded_size1 %" PRIu64 "\n", size1, padded_size1);
```

Comment:
Example of the `printf` format for `uint64_t`. I totally agree it's not very beautiful, and I'm happy to revert if that's wanted.